src/content/posts/2023-10-24-git-workflow.md

```diff
@@ -0,0 +1,132 @@
+
+### Managing repositories
+
+##### The commands
+- `git clone {url}` - clones down a respository
+- `git remote add {name} {url}` - sets a new remote with the given name
+
+##### The workflow
+- Fork the repository from your organization to your own github account
+- Clone down your fork of the repository using `git clone {fork url}`
+- Add your organization's copy of the repository as a remote with the desired name (we always use `upstream`) using `git remote add {name} {main repo url}`
+
+##### The general idea
+
+You should always have your own fork of your organization's repository that you can do your work from. This isolates your work from the organization's codebase which adds a layer of protection. If you use `upstream` as the name of the organization's copy of the repository, you'll end up with two remotes (the codebase stored on github) to manage. You'll have `origin` which is your copy and you'll have `upstream` which is the organization's copy.
+
+### Navigating branches
+
+##### The commands
+- `git checkout {branch}` - moves to an existing branch of the provided name
+- `git checkout -b {branch}` - creates a new branch with the provided name, using the current branch as the base
+- `git branch -d {branch}` - deletes the branch with the provided name from your local machine
+- `git push origin :{branch}` - deletes the branch from your github
+
+##### The workflow
+- When creating a new feature branch, you'll want to create it off of your master (or feature) branch, like so:
+```
+git checkout master
+git checkout -b my-feature
+```
+- When cleaning up old branches, you can get a look at your branches with `git remote show origin` and then delete them with the above commands, like so:
+```
+git remote show origin
+git branch -d my-feature
+git push origin :my-feature
+```
+
+##### The general idea
+
+Your work should never be done on your own master branch. Your work should be done on a branch that you've created using your master branch (once it's up to date with the upstream - keep reading for that section!) as a base. These commands will help with moving between branches, creating new branches, and cleaning up stale branches. You can view your existing branches with `git remote show origin` which will show the branches in your fork.
+
+### Contributing to codebases
+
+##### The commands
+- `git add .`
+- `git commit -m "message"`
+- `git push -u origin {branch}`
```

Comment:
Should I be pushing with the `-u` in there?

Reply:
It just makes it so if you do just `git push` it'll go to the branch you specified with the `-u` in there

Reply:
Dude, your post is already working before it was even live. I've always done `git push origin {my-branch}`. I'm learning so much.